sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -372,6 +378,51 @@ public EventHubClientBuilder prefetchCount(int prefetchCount) {
         return this;
     }
 
+    /**
+     * Enables idempotent publishing when an {@link EventHubProducerAsyncClient} or {@link EventHubProducerClient}
+     * is built.
+     *
+     * If enabled, the producer will only be able to publish directly to partitions; it will not be able to publish to
+     * the Event Hubs gateway for automatic partition routing nor using a partition key.
+     *
+     * @return The updated {@link EventHubClientBuilder} object.
+     */
+    public EventHubClientBuilder idempotentPartitionPublishing() {
+        this.idempotentPartitionPublishing = true;
+        return this;
+    }
+
+    /**
+     * Sets the idempotent publishing options to {@link EventHubProducerAsyncClient} or {@link EventHubProducerClient}
+     * when you build them.
+     *
+     * The set of options that can be specified to influence publishing behavior specific to the configured Event Hub
+     * partition.
+     * These options are not necessary in the majority of scenarios and are intended for use with specialized scenarios,
+     * such as when recovering the state used for idempotent publishing.
+     *
+     * It is highly recommended that these options only be specified if there is a proven need to do so; Incorrectly
+     * configuring these values may result in the built {@link EventHubProducerAsyncClient} or
+     * {@link EventHubProducerClient} instance unable to publish to the Event Hubs.
+     *
+     * These options are ignored when publishing to the Event Hubs gateway for automatic routing or when using a
+     * partition key.
+     *
+     * @param states A {@link Map} of {@link PartitionPublishingState} for each partition. The keys of the map
+     * are the partition ids.
+     * @return The updated {@link EventHubClientBuilder} object.
+     */
+    public EventHubClientBuilder initialPartitionPublishingStates(Map<String,
+        PartitionPublishingProperties> states) {
```

Comment:
Can we put the entire argument on the next line rather than splitting the map here?

Reply:
Updated. I forget what happened on me to make it two lines ;-)